ssr+i18n throw In-memory ESM relative file should always exist

### Command

build

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

If i18n, SSR, and assets request are configured, the following error occurs.

config
```json
{
          "builder": "@angular/build:application",
          "options": {
            "localize": [
              "en"
            ],
            "browser": "src/main.ts",
            "tsConfig": "tsconfig.app.json",
            "assets": [
              {
                "glob": "**/*",
                "input": "public"
              }
            ],
            "styles": [
              "src/styles.css"
            ],
            "server": "src/main.server.ts",
            "outputMode": "server",
            "security": {
              "allowedHosts": []
            },
            "ssr": {
              "entry": "src/server.ts"
            }
          }
```
and request any assets
```
 data = resource({
    loader: async (input) => {
      return firstValueFrom(this.#http.get(`test.text`, { responseType: 'text' }));
    },
  });
```
build log as follow
```log
\➜  ssr1 git:(master) ✗npm run build
Debugger listening on ws://127.0.0.1:62952/ba55e689-9694-4a67-a90c-4fa2f7ca3f78
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
npm warn Unknown user config "msvs_version". This will stop working in the next major version of npm.

> ssr1@0.0.0 build
> ng build        

Debugger listening on ws://127.0.0.1:62954/259aa559-5bf5-40e1-8f32-6ccece67e40d
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.

uncaughtException AssertionError [ERR_ASSERTION]: In-memory ESM relative file should always exist: 'file:///C:/code/my-project/ssr1/.angular/prerender-root/34778af1-d6e3-4e7e-85f6-68a2c814c5b3/angular-app-engine-manifest.mjs' --> './en/main.server.mjs'
    at resolve (C:\code\my-project\ssr1\node_modules\@angular\build\src\utils\server-rendering\esm-in-memory-loader\loader-hooks.js:80:31)       
    at nextResolve (node:internal/modules/esm/hooks:748:28)
    at Hooks.resolve (node:internal/modules/esm/hooks:240:30)
    at MessagePort.handleMessage (node:internal/modules/esm/worker:201:24)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:845:20)
    at MessagePort.<anonymous> (node:internal/per_context/messageport:23:28)
    at MessagePort.callbackTrampoline (node:internal/async_hooks:130:17) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: undefined,
  expected: undefined,
  operator: 'fail',
  diff: 'simple'
}
Browser bundles
Initial chunk files   | Names            |  Raw size | Estimated transfer size
main-EQM76B7T.js      | main             | 249.94 kB |                68.45 kB
polyfills-7FR7WZWP.js | polyfills        | 136 bytes |               136 bytes
styles-5INURTSO.css   | styles           |   0 bytes |                 0 bytes

                      | Initial total    | 250.07 kB |                68.59 kB


Server bundles
Initial chunk files   | Names            |  Raw size
server.mjs            | server           | 810.29 kB |
main.server.mjs       | main.server      | 684.96 kB |
polyfills.server.mjs  | polyfills.server | 233.39 kB |
chunk-WVQMRNL5.mjs    | -                |   1.39 kB |
chunk-VB3FBMFV.mjs    | -                |   1.05 kB |

Lazy chunk files      | Names            |  Raw size
chunk-HPMYPLCO.mjs    | xhr2             |  12.22 kB |
chunk-OUZZXTQV.mjs    | xhr2             |  12.15 kB |

Prerendered 1 static route.
Application bundle generation failed. [38.307 seconds] - 2026-03-06T08:40:11.923Z

✘ [ERROR] An error occurred while prerendering route '/aa/en/test1'.

Request for: http://localhost:62982/aa/en/test1 was aborted.
TimeoutError: The operation was aborted due to timeout
```

### Minimal Reproduction

clone https://github.com/wszgrcy/ssr-issue
npm i
npm start
go to http://localhost:4212/

### Exception or Error

```text

```

### Your Environment

```text
Angular CLI       : 21.2.0
Angular           : 21.2.1
Node.js           : 24.10.0
Package Manager   : npm 11.6.1
Operating System  : win32 x64

┌───────────────────────────┬───────────────────┬───────────────────┐
│ Package                   │ Installed Version │ Requested Version │
├───────────────────────────┼───────────────────┼───────────────────┤
│ @angular/build            │ 21.2.0            │ ^21.2.0           │
│ @angular/cli              │ 21.2.0            │ ^21.2.0           │
│ @angular/common           │ 21.2.1            │ ^21.2.0           │
│ @angular/compiler         │ 21.2.1            │ ^21.2.0           │
│ @angular/compiler-cli     │ 21.2.1            │ ^21.2.0           │
│ @angular/core             │ 21.2.1            │ ^21.2.0           │
│ @angular/forms            │ 21.2.1            │ ^21.2.0           │
│ @angular/localize         │ 21.2.1            │ ^21.2.1           │
│ @angular/platform-browser │ 21.2.1            │ ^21.2.0           │
│ @angular/platform-server  │ 21.2.1            │ ^21.2.0           │
│ @angular/router           │ 21.2.1            │ ^21.2.0           │
│ @angular/ssr              │ 21.2.0            │ ^21.2.0           │
│ rxjs                      │ 7.8.2             │ ~7.8.0            │
│ typescript                │ 5.9.3             │ ~5.9.2            │
│ vitest                    │ 4.0.18            │ ^4.0.8            │
```

### Anything else relevant?

_No response_

--- a/packages/angular/build/src/utils/server-rendering/prerender.ts
+++ b/packages/angular/build/src/utils/server-rendering/prerender.ts
@@ -96,7 +96,9 @@ export async function prerenderPages(
 
   const assetsReversed: Record</** Destination */ string, /** Source */ string> = {};
   for (const { source, destination } of assets) {
-    assetsReversed[addLeadingSlash(toPosixPath(destination))] = source;
+    // Assets are not stored with baseHref when using i18n,
+    // we append the base href so that requests are resolved correctly.
+    assetsReversed[joinUrlParts(baseHref, toPosixPath(destination))] = source;
   }
 
   // Get routes to prerender

--- a/tests/e2e/tests/build/server-rendering/server-routes-output-mode-static-i18n-http-calls.ts
+++ b/tests/e2e/tests/build/server-rendering/server-routes-output-mode-static-i18n-http-calls.ts
@@ -1,0 +1,137 @@
+import assert, { match } from 'node:assert';
+import { join } from 'node:path';
+import { readFile, writeMultipleFiles } from '../../../utils/fs';
+import { ng, noSilentNg, silentNg } from '../../../utils/process';
+import { getGlobalVariable } from '../../../utils/env';
+import { installWorkspacePackages, uninstallPackage } from '../../../utils/packages';
+import { useSha } from '../../../utils/project';
+import { langTranslations, setupI18nConfig } from '../../i18n/setup';
+
+export default async function () {
+  assert(
+    getGlobalVariable('argv')['esbuild'],
+    'This test should not be called in the Webpack suite.',
+  );
+
+  // Setup project
+  await setupI18nConfig();
+
+  // Forcibly remove in case another test doesn't clean itself up.
+  await uninstallPackage('@angular/ssr');
+  await ng('add', '@angular/ssr', '--skip-confirmation', '--skip-install');
+  await useSha();
+  await installWorkspacePackages();
+
+  await writeMultipleFiles({
+    // Add asset
+    'public/media.json': JSON.stringify({ dataFromAssets: true }),
+    // Update component to do an HTTP call to asset and API.
+    'src/app/app.ts': `
+    import { ChangeDetectorRef, Component, inject } from '@angular/core';
+    import { JsonPipe } from '@angular/common';
+    import { RouterOutlet } from '@angular/router';
+    import { HttpClient } from '@angular/common/http';
+
+    @Component({
+      selector: 'app-root',
+      imports: [JsonPipe, RouterOutlet],
+      template: \`
+        <p>{{ assetsData | json }}</p>
+        <p>{{ apiData | json }}</p>
+        <router-outlet></router-outlet>
+      \`,
+    })
+    export class App {
+      assetsData: any;
+      apiData: any;
+      private readonly cdr: ChangeDetectorRef = inject(ChangeDetectorRef);
+
+      constructor() {
+        const http = inject(HttpClient);
+
+        http.get('media.json').toPromise().then((d) => {
+          this.assetsData = d;
+          this.cdr.markForCheck();
+        });
+
+        http.get('/api').toPromise().then((d) => {
+          this.apiData = d;
+          this.cdr.markForCheck();
+        });
+      }
+    }
+    `,
+    // Add http client and route
+    'src/app/app.config.ts': `
+      import { ApplicationConfig } from '@angular/core';
+      import { provideRouter } from '@angular/router';
+
+      import { Home } from './home/home';
+      import { provideClientHydration } from '@angular/platform-browser';
+      import { provideHttpClient, withFetch } from '@angular/common/http';
+
+      export const appConfig: ApplicationConfig = {
+        providers: [
+          provideRouter([{
+            path: 'home',
+            component: Home,
+          }]),
+          provideClientHydration(),
+          provideHttpClient(withFetch()),
+        ],
+      };
+    `,
+    'src/server.ts': `
+      import { AngularNodeAppEngine, writeResponseToNodeResponse, isMainModule, createNodeRequestHandler } from '@angular/ssr/node';
+      import express from 'express';
+      import { join } from 'node:path';
+
+      export function app(): express.Express {
+        const server = express();
+        const browserDistFolder = join(import.meta.dirname, '../browser');
+        const angularNodeAppEngine = new AngularNodeAppEngine();
+
+        server.get('/api', (req, res) => {
+          res.json({ dataFromAPI: true })
+        });
+
+        server.use(express.static(browserDistFolder, {
+          maxAge: '1y',
+          index: 'index.html'
+        }));
+
+        server.use((req, res, next) => {
+          angularNodeAppEngine.handle(req)
+            .then((response) => response ? writeResponseToNodeResponse(response, res) : next())
+            .catch(next);
+        });
+        return server;
+      }
+
+      const server = app();
+
+      if (isMainModule(import.meta.url)) {
+        const port = process.env['PORT'] || 4000;
+        server.listen(port, (error) => {
+          if (error) {
+            throw error;
+          }
+          console.log(\`Node Express server listening on http://localhost:\${port}\`);
+        });
+      }
+
+      export const reqHandler = createNodeRequestHandler(server);
+    `,
+  });
+
+  await silentNg('generate', 'component', 'home');
+
+  await noSilentNg('build', '--output-mode=static');
+
+  for (const { lang, outputPath } of langTranslations) {
+    const contents = await readFile(join(outputPath, 'home/index.html'));
+    match(contents, /<p>{[\S\s]*"dataFromAssets":[\s\S]*true[\S\s]*}<\/p>/);
+    match(contents, /<p>{[\S\s]*"dataFromAPI":[\s\S]*true[\S\s]*}<\/p>/);
+    match(contents, new RegExp(`<base href="\\/${lang}\\/">`));
+  }
+}

